src/syntax/syntax.rkt

```diff
@@ -381,10 +387,55 @@
                            (define-values (_ exs) (real-apply compiler pt))
                            (if exs (first exs) fail))
                          name)]))
+
+  ; make hash table
+  (define rules (make-hasheq))
+  (define count 0)
+  (define commutes? #f)
+  (when identities
+    (for ([ident (in-list identities)])
+      (match ident
+        [(list ident-name lhs-expr rhs-expr)
+         (cond
+           [(hash-has-key? rules ident-name)
+            (raise-herbie-syntax-error "Duplicate identity ~a" ident-name)]
+           [else
+            (hash-set! rules
+                       (string->symbol (format "~a-~a" (symbol->string ident-name) name))
+                       (list lhs-expr
+                             rhs-expr
+                             (remove-duplicates (append (free-variables lhs-expr)
+                                                        (free-variables rhs-expr)))))])]
+        [(list 'exact expr)
+         (hash-set! rules
+                    (gensym (string->symbol (format "~a-exact-~a" name count)))
+                    (list expr expr (free-variables expr)))
+         (set! count (+ count 1))]
```

Comment:
It's unclear to me how this is generating an exact rule. Isn't this just saying `x => x`?